hack/loadtest/util/placement.go

```diff
@@ -79,148 +93,227 @@ func MeasureOnePlacement(ctx context.Context, hubClient client.Client, deadline,
 	startTime := time.Now()
 	applyDeadline := startTime.Add(deadline)
-	var crp v1alpha1.ClusterResourcePlacement
+	var crp v1beta1.ClusterResourcePlacement
 	var err error
-	for ; ctx.Err() == nil; time.Sleep(pollInterval) {
-		if err = hubClient.Get(ctx, types.NamespacedName{Name: crpName, Namespace: ""}, &crp); err != nil {
-			klog.ErrorS(err, "failed to get crp", "crp", crpName)
-			if time.Now().After(applyDeadline) {
-				// timeout
-				klog.V(2).Infof("the cluster resource placement `%s` timeout", crpName)
-				LoadTestApplyCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
-				applyTimeoutCount.Inc()
-				break
+	ticker := time.NewTicker(pollInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return fleetSize, clusterNames
+		case <-ticker.C:
+			if err = hubClient.Get(ctx, types.NamespacedName{Name: crpName, Namespace: ""}, &crp); err != nil {
+				klog.ErrorS(err, "failed to get crp", "crp", crpName)
+				if time.Now().After(applyDeadline) {
+					// timeout
+					klog.V(2).Infof("the cluster resource placement `%s` timeout", crpName)
+					LoadTestApplyCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
+					applyTimeoutCount.Inc()
+					return fleetSize, clusterNames
+				}
+				continue
 			}
-			continue
-		}
-		// check if the condition is true
-		cond := crp.GetCondition(string(v1alpha1.ResourcePlacementStatusConditionTypeApplied))
-		if cond == nil || cond.Status == metav1.ConditionUnknown {
-			klog.V(5).Infof("the cluster resource placement `%s` is pending", crpName)
-		} else if cond != nil && cond.Status == metav1.ConditionTrue {
-			// succeeded
-			klog.V(3).Infof("the cluster resource placement `%s` succeeded", crpName)
-			LoadTestApplyCountMetric.WithLabelValues(currency, fleetSize, "succeed").Inc()
-			applySuccessCount.Inc()
-			LoadTestApplyLatencyMetric.WithLabelValues(currency, fleetSize).Observe(time.Since(startTime).Seconds())
-			break
-		}
-		if time.Now().After(applyDeadline) {
-			if cond != nil && cond.Status == metav1.ConditionFalse {
-				// failed
-				klog.V(2).Infof("the cluster resource placement `%s` failed", crpName)
-				LoadTestApplyCountMetric.WithLabelValues(currency, fleetSize, "failed").Inc()
-				applyFailCount.Inc()
-			} else {
-				// timeout
-				klog.V(2).Infof("the cluster resource placement `%s` timeout", crpName)
-				LoadTestApplyCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
-				applyTimeoutCount.Inc()
+			// check if the condition is true
+			cond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementAppliedConditionType))
+			if cond == nil || cond.Status == metav1.ConditionUnknown {
+				klog.V(5).Infof("the cluster resource placement `%s` is pending", crpName)
+			} else if cond != nil && cond.Status == metav1.ConditionTrue {
```

Comment:
Hi Britania! This can be a `IsConditionStatusTrue` call I believe. The check above can be in the `else` branch I think.